src/openrct2/ride/Vehicle.cpp

```diff
@@ -7251,18 +7255,20 @@ static void vehicle_update_additional_animation(rct_vehicle* vehicle)
                             || (vehicle->status != VEHICLE_STATUS_MOVING_TO_END_OF_STATION
                                 && vehicle->status != VEHICLE_STATUS_ARRIVING))
                         {
-                            int32_t index = vehicle->sprite_direction >> 1;
-                            if (vehicle->vehicle_sprite_type == 2)
-                            {
-                                index += 16;
-                            }
-                            if (vehicle->vehicle_sprite_type == 6)
-                            {
-                                index += 32;
-                            }
-                            steam_particle_create(
-                                vehicle->x + SteamParticleOffsets[index].x, vehicle->y + SteamParticleOffsets[index].y,
-                                vehicle->z + SteamParticleOffsets[index].z);
+                            int32_t typeIndex = [&] {
+                                switch (vehicle->vehicle_sprite_type)
+                                {
+                                    case 2:
```

Comment:
Do we have any constants for these yet? If not, there really should be.

Reply:
Case 2 is for going uphill, case 6 is for going downhill. I don't think there are constants for these though. It would require quite a lot of investigation to work out. I think for now just leave a comment stating that case 2 is uphill and case 6 is downhill.